mongo/client_examples_test.go

```diff
@@ -423,3 +423,42 @@ func ExampleConnect_stableAPI() {
 	}
 	_ = serverAPIDeprecationClient
 }
+
+func ExampleConnect_bSONOptions() {
+	// Configure a client that customizes the BSON marshal and unmarshal
+	// behavior.
+
+	clientOpts := options.Client().
+		ApplyURI("mongodb://localhost:27017").
+		// Specify BSON options that cause the driver to fallback to "json"
+		// struct tags if "bson" struct tags are missing, marshal nil Go maps as
+		// empty BSON documents, and marshals nil Go slices as empty BSON
+		// arrays.
+		SetBSONOptions(&options.BSONOptions{
+			UseJSONStructTags: true,
+			NilMapAsEmpty:     true,
+			NilSliceAsEmpty:   true,
```

Comment:
```suggestion
		SetBSONOptions(&options.BSONOptions{
			UseJSONStructTags: true, // fallback to "json" tags if "bson" tags are missing
			NilMapAsEmpty:     true, // marshal nil Go maps as empty BSON documents
			NilSliceAsEmpty:   true, // marshal nil Go slices as empty BSON arrays
```

The comment break between setters is sort of hard to read.

Reply:
I'm concerned the formatting of same-line comments will get mangled when rendered in the `pkg.go.dev` docs site. However, I can move the `BSONOptions` struct definition above the setters to make it cleaner.

Reply:
No it's okay, thank you for checking